cio/src/providers.rs

```diff
@@ -177,21 +206,19 @@ impl ProviderOps<ramp_api::types::User, ()> for ramp_api::Client {
     async fn list_provider_groups(&self, _company: &Company) -> Result<Vec<()>> {
         Ok(vec![])
     }
-
-    async fn delete_user(&self, _db: &Database, _company: &Company, _user: &User) -> Result<()> {
-        // TODO: Suspend the user from Ramp.
-        Ok(())
-    }
-
-    // Ramp does not have groups so this is a no-op.
-    async fn delete_group(&self, _company: &Company, _group: &Group) -> Result<()> {
-        Ok(())
-    }
 }
 
 #[async_trait]
-impl ProviderOps<octorust::types::SimpleUser, octorust::types::Team> for octorust::Client {
+impl ProviderWriteOps for octorust::Client {
     async fn ensure_user(&self, _db: &Database, company: &Company, user: &User) -> Result<String> {
+        if user.denied_services.contains(&ExternalServices::GitHub) {
+            log::info!(
+                "User {} is denied access to {}. Will exit provisioning when deny lists are enforced.",
+                user.id,
+                ExternalServices::GitHub
+            );
+        }
```

Comment:
Should these be returning early or something? I'm not sure if I'm missing where the denial is actually happening or if this is just a first pass to log what should happen for testing.